src/utils/routes/route.js

```diff
@@ -0,0 +1,18 @@
+import genericController from './genericController';
+
+class Route {
+  static factory(method, route, controller, controllerMethod, permission) {
+    return router =>
```

Comment:
I think this extra closure can be avoided.

Route.factory (and each of the methods) can create an object, instead of a function likeso:

```
get = (route, controller, method, permission) => ({
   method: 'get'
   route, 
   controller,
   method, 
   permission
});
```

Now your addRoutes method will accept a router and an array of objects, instead of the array of functions so then you can do
```
routes.forEach(routeConfig => router[routeConfig.method](routeConfig.route, genericController(...))
```

Reply:
It was my firts variat... xD
Unfortunately I didn't commit it...